docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+
+function Cat(props) {
+  return (
+    <View>
+      <Text>Hello, I am {props.name}!</Text>
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <View>
+      <Cat name="Maru" />
+      <Cat name="Jellylorum" />
+      <Cat name="Spot" />
+    </View>
+  );
+}
+```
+
+Most of React Native’s Core Components can be customized with props, too. For example, when using [`Image`](image), you pass it a `prop` named [`source`](image#source) to define what image it shows:
+
+```SnackPlayer name=Props
+import React from 'react';
+import { Text, View, Image } from 'react-native';
+
+export default function CatApp() {
+  return (
+    <View>
+      <Image
+        source='https://placekitten.com/g/193/110'
+        style={{
+          width: 193,
+          height: 110
+        }}/>
+      <Text>Hello, I am your cat!</Text>
+    </View>
+  );
+}
+```
+
+`Image` has [many different props](image#props), including [`style`](image#style), which accepts a JS object of design and layout related property-value pairs.
+
+> Notice the double curly braces `{{ }}` surrounding `style`‘s width and height. In JSX, JavaScript values are referenced with `{}`. This is handy if you are passing something other than a string as props, like an array or number (`<Cat food={["fish", "kibble"]} /> age={2}`). However, JS object are **_also_** denoted with curly braces: `{width: 193, height: 110}`. Therefore, to pass a JS object in JSX, you must wrap the object in **another pair** of curly braces: `{{width: 193, height: 110}}`
+
+You can add custom cat pictures to your component by:
+
+- adding an `Image` component to your `Cat` component
+- setting `Image`‘s `source` via `Cat`‘s `props` with `{props.picURI}`
+- then passing `picURI` values in each `Cat`
+
+```SnackPlayer name=Custom%20Pictures
+import React from 'react';
+import { Text, View, Image } from 'react-native';
+
+function Cat(props) {
+  return (
+    <View>
+      <Image source={props.picURI} style={{width: 200, height: 200}}/>
+      <Text>Hello, I am {props.name}!</Text>
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <View>
+      <Cat name="Maru" picURI="https://placekitten.com/g/200/200" />
+      <Cat name="Jellylorum" picURI="https://placekitten.com/g/200/200" />
+      <Cat name="Spot" picURI="https://placekitten.com/g/200/200" />
+    </View>
+  );
+}
+```
+
+You can build many things with `props` and the Core Components [`Text`](text), [`Image`](image), and [`View`](view)! But to build something interactive, you’ll need `state`.
+
+## State
+
+Props are set by the parent component and can only be changed by the parent component passing down new values. But for individual components to change according to human interactions, they need to be able to accept and store their own data. Fortunately, React components can have their own `state` you can get and set data from.
+
```

Comment:
I find that this paragraph still leaves me fuzzy about the distinction between props and state. Maybe some analogy could work? Props are like "component arguments", and state is like "component variables". State is always owned by a particular component, which can pass it down as props to other components.

Reply:
Let me have a think! I also get tripped up on this one. Maybe I can see how others are shorthanding it to their students...

Reply:
How about "While you can think of props as arguments you use to configure how components render, state is like a component’s personal data storage. Sate is useful for handling data that changes over time or that comes from user interaction."